### ISSUE TITLE
zfs-dkms not installing properly

### ISSUE THREAD

**Issue by aimforthebushes** [pop-os/pop#1659]
**Distribution (run `cat /etc/os-release`):**
 Today I was purging a package (teamviewer) for my system (Pop_OS 20.04) and for some reason apt begin to try to remove all zfs-dpkg packages and throuhs an error:

sudo apt purge teamviewer
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libqt5positioning5 libqt5sensors5 libqt5webchannel5 libqt5webkit5 linux-headers-5.8.0-7630
  linux-headers-5.8.0-7630-generic linux-image-5.8.0-7630-generic
  linux-modules-5.8.0-7630-generic linux-modules-extra-5.8.0-7630-generic
  qml-module-qtquick-controls qml-module-qtquick-dialogs qml-module-qtquick-privatewidgets
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  teamviewer*
0 upgraded, 0 newly installed, 1 to remove and 132 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] y
Setting up zfs-dkms (0.8.3-1ubuntu12.7) ...
Removing old zfs-0.8.3 DKMS files...

------------------------------
Deleting module version: 0.8.3
completely from the DKMS tree.
------------------------------
Done.
Loading new zfs-0.8.3 DKMS files...
Building for 5.11.0-7612-generic
Building initial module for 5.11.0-7612-generic
configure: error: 
	*** Unable to build an empty module.
			
ERROR (dkms apport): kernel package linux-headers-5.11.0-7612-generic is not supported
Error! Bad return status for module build on kernel: 5.11.0-7612-generic (x86_64)
Consult /var/lib/dkms/zfs/0.8.3/build/make.log for more information.
dpkg: error processing package zfs-dkms (--configure):
 installed zfs-dkms package post-installation script subprocess returned error exit status 10
Processing triggers for initramfs-tools (0.136ubuntu6.4) ...
update-initramfs: Generating /boot/initrd.img-5.11.0-7612-generic
kernelstub.Config    : INFO     Looking for configuration...
kernelstub           : INFO     System information: 

    OS:..................Pop!_OS 20.04
    Root partition:....../dev/nvme0n1p3
    Root FS UUID:........7420ff3c-9406-4498-b619-de71661dc142
    ESP Path:............/boot/efi
    ESP Partition:......./dev/nvme0n1p1
    ESP Partition #:.....1
    NVRAM entry #:.......-1
    Boot Variable #:.....0000
    Kernel Boot Options:.quiet loglevel=0 systemd.show_status=false splash
    Kernel Image Path:.../boot/vmlinuz-5.11.0-7612-generic
    Initrd Image Path:.../boot/initrd.img-5.11.0-7612-generic
    Force-overwrite:.....False

kernelstub.Installer : INFO     Copying Kernel into ESP
kernelstub.Installer : INFO     Copying initrd.img into ESP
kernelstub.Installer : INFO     Setting up loader.conf configuration
kernelstub.Installer : INFO     Making entry file for Pop!_OS
kernelstub.Installer : INFO     Backing up old kernel
kernelstub.Installer : INFO     Making entry file for Pop!_OS
Errors were encountered while processing:
 zfs-dkms
E: Sub-process /usr/bin/dpkg returned an error code (1)

Then I've decided to purge zfs-dkms and after trying to install it again I get the following:

sudo apt purge teamviewer
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libqt5positioning5 libqt5sensors5 libqt5webchannel5 libqt5webkit5 linux-headers-5.8.0-7630
  linux-headers-5.8.0-7630-generic linux-image-5.8.0-7630-generic
  linux-modules-5.8.0-7630-generic linux-modules-extra-5.8.0-7630-generic
  qml-module-qtquick-controls qml-module-qtquick-dialogs qml-module-qtquick-privatewidgets
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  teamviewer*
0 upgraded, 0 newly installed, 1 to remove and 132 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] y
Setting up zfs-dkms (0.8.3-1ubuntu12.7) ...
Removing old zfs-0.8.3 DKMS files...

------------------------------
Deleting module version: 0.8.3
completely from the DKMS tree.
------------------------------
Done.
Loading new zfs-0.8.3 DKMS files...
Building for 5.11.0-7612-generic
Building initial module for 5.11.0-7612-generic
configure: error: 
	*** Unable to build an empty module.
			
ERROR (dkms apport): kernel package linux-headers-5.11.0-7612-generic is not supported
Error! Bad return status for module build on kernel: 5.11.0-7612-generic (x86_64)
Consult /var/lib/dkms/zfs/0.8.3/build/make.log for more information.
dpkg: error processing package zfs-dkms (--configure):
 installed zfs-dkms package post-installation script subprocess returned error exit status 10
Processing triggers for initramfs-tools (0.136ubuntu6.4) ...
update-initramfs: Generating /boot/initrd.img-5.11.0-7612-generic
kernelstub.Config    : INFO     Looking for configuration...
kernelstub           : INFO     System information: 

    OS:..................Pop!_OS 20.04
    Root partition:....../dev/nvme0n1p3
    Root FS UUID:........7420ff3c-9406-4498-b619-de71661dc142
    ESP Path:............/boot/efi
    ESP Partition:......./dev/nvme0n1p1
    ESP Partition #:.....1
    NVRAM entry #:.......-1
    Boot Variable #:.....0000
    Kernel Boot Options:.quiet loglevel=0 systemd.show_status=false splash
    Kernel Image Path:.../boot/vmlinuz-5.11.0-7612-generic
    Initrd Image Path:.../boot/initrd.img-5.11.0-7612-generic
    Force-overwrite:.....False

kernelstub.Installer : INFO     Copying Kernel into ESP
kernelstub.Installer : INFO     Copying initrd.img into ESP
kernelstub.Installer : INFO     Setting up loader.conf configuration
kernelstub.Installer : INFO     Making entry file for Pop!_OS
kernelstub.Installer : INFO     Backing up old kernel
kernelstub.Installer : INFO     Making entry file for Pop!_OS
Errors were encountered while processing:
 zfs-dkms
E: Sub-process /usr/bin/dpkg returned an error code (1)

tldr;

Building initial module for 5.11.0-7612-generic
configure: error: 
	*** Unable to build an empty module.
			
ERROR (dkms apport): kernel package linux-headers-5.11.0-7612-generic is not supported
Error! Bad return status for module build on kernel: 5.11.0-7612-generic (x86_64)
Consult /var/lib/dkms/zfs/0.8.3/build/make.log for more information.
dpkg: error processing package zfs-dkms (--configure):
 installed zfs-dkms package post-installation script subprocess returned error exit status 10

What can I do to be able to use zfs?
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
zfs-dkms:
  Installed: 0.8.3-1ubuntu12.7
  Candidate: 0.8.3-1ubuntu12.7
  Version table:
 *** 0.8.3-1ubuntu12.7 500
        500 http://us.archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages
        500 http://us.archive.ubuntu.com/ubuntu focal-updates/universe i386 Packages
        100 /var/lib/dpkg/status
     0.8.3-1ubuntu12 500
        500 http://us.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
        500 http://us.archive.ubuntu.com/ubuntu focal/universe i386 Packages

**Issue/Bug Description:**
Today I was purging a package (teamviewer) for my system (Pop_OS 20.04) and for some reason apt begin to try to remove all zfs-dpkg packages and throuhs an error:

```
sudo apt purge teamviewer
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libqt5positioning5 libqt5sensors5 libqt5webchannel5 libqt5webkit5 linux-headers-5.8.0-7630
  linux-headers-5.8.0-7630-generic linux-image-5.8.0-7630-generic
  linux-modules-5.8.0-7630-generic linux-modules-extra-5.8.0-7630-generic
  qml-module-qtquick-controls qml-module-qtquick-dialogs qml-module-qtquick-privatewidgets
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  teamviewer*
0 upgraded, 0 newly installed, 1 to remove and 132 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] y
Setting up zfs-dkms (0.8.3-1ubuntu12.7) ...
Removing old zfs-0.8.3 DKMS files...

------------------------------
Deleting module version: 0.8.3
completely from the DKMS tree.
------------------------------
Done.
Loading new zfs-0.8.3 DKMS files...
Building for 5.11.0-7612-generic
Building initial module for 5.11.0-7612-generic
configure: error: 
	*** Unable to build an empty module.
			
ERROR (dkms apport): kernel package linux-headers-5.11.0-7612-generic is not supported
Error! Bad return status for module build on kernel: 5.11.0-7612-generic (x86_64)
Consult /var/lib/dkms/zfs/0.8.3/build/make.log for more information.
dpkg: error processing package zfs-dkms (--configure):
 installed zfs-dkms package post-installation script subprocess returned error exit status 10
Processing triggers for initramfs-tools (0.136ubuntu6.4) ...
update-initramfs: Generating /boot/initrd.img-5.11.0-7612-generic
kernelstub.Config    : INFO     Looking for configuration...
kernelstub           : INFO     System information: 

    OS:..................Pop!_OS 20.04
    Root partition:....../dev/nvme0n1p3
    Root FS UUID:........7420ff3c-9406-4498-b619-de71661dc142
    ESP Path:............/boot/efi
    ESP Partition:......./dev/nvme0n1p1
    ESP Partition #:.....1
    NVRAM entry #:.......-1
    Boot Variable #:.....0000
    Kernel Boot Options:.quiet loglevel=0 systemd.show_status=false splash
    Kernel Image Path:.../boot/vmlinuz-5.11.0-7612-generic
    Initrd Image Path:.../boot/initrd.img-5.11.0-7612-generic
    Force-overwrite:.....False

kernelstub.Installer : INFO     Copying Kernel into ESP
kernelstub.Installer : INFO     Copying initrd.img into ESP
kernelstub.Installer : INFO     Setting up loader.conf configuration
kernelstub.Installer : INFO     Making entry file for Pop!_OS
kernelstub.Installer : INFO     Backing up old kernel
kernelstub.Installer : INFO     Making entry file for Pop!_OS
Errors were encountered while processing:
 zfs-dkms
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Then I've decided to purge zfs-dkms and after trying to install it again I get the following:
```

sudo apt purge teamviewer
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libqt5positioning5 libqt5sensors5 libqt5webchannel5 libqt5webkit5 linux-headers-5.8.0-7630
  linux-headers-5.8.0-7630-generic linux-image-5.8.0-7630-generic
  linux-modules-5.8.0-7630-generic linux-modules-extra-5.8.0-7630-generic
  qml-module-qtquick-controls qml-module-qtquick-dialogs qml-module-qtquick-privatewidgets
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  teamviewer*
0 upgraded, 0 newly installed, 1 to remove and 132 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] y
Setting up zfs-dkms (0.8.3-1ubuntu12.7) ...
Removing old zfs-0.8.3 DKMS files...

------------------------------
Deleting module version: 0.8.3
completely from the DKMS tree.
------------------------------
Done.
Loading new zfs-0.8.3 DKMS files...
Building for 5.11.0-7612-generic
Building initial module for 5.11.0-7612-generic
configure: error: 
	*** Unable to build an empty module.
			
ERROR (dkms apport): kernel package linux-headers-5.11.0-7612-generic is not supported
Error! Bad return status for module build on kernel: 5.11.0-7612-generic (x86_64)
Consult /var/lib/dkms/zfs/0.8.3/build/make.log for more information.
dpkg: error processing package zfs-dkms (--configure):
 installed zfs-dkms package post-installation script subprocess returned error exit status 10
Processing triggers for initramfs-tools (0.136ubuntu6.4) ...
update-initramfs: Generating /boot/initrd.img-5.11.0-7612-generic
kernelstub.Config    : INFO     Looking for configuration...
kernelstub           : INFO     System information: 

    OS:..................Pop!_OS 20.04
    Root partition:....../dev/nvme0n1p3
    Root FS UUID:........7420ff3c-9406-4498-b619-de71661dc142
    ESP Path:............/boot/efi
    ESP Partition:......./dev/nvme0n1p1
    ESP Partition #:.....1
    NVRAM entry #:.......-1
    Boot Variable #:.....0000
    Kernel Boot Options:.quiet loglevel=0 systemd.show_status=false splash
    Kernel Image Path:.../boot/vmlinuz-5.11.0-7612-generic
    Initrd Image Path:.../boot/initrd.img-5.11.0-7612-generic
    Force-overwrite:.....False

kernelstub.Installer : INFO     Copying Kernel into ESP
kernelstub.Installer : INFO     Copying initrd.img into ESP
kernelstub.Installer : INFO     Setting up loader.conf configuration
kernelstub.Installer : INFO     Making entry file for Pop!_OS
kernelstub.Installer : INFO     Backing up old kernel
kernelstub.Installer : INFO     Making entry file for Pop!_OS
Errors were encountered while processing:
 zfs-dkms
E: Sub-process /usr/bin/dpkg returned an error code (1)

```
tldr;

```
Building initial module for 5.11.0-7612-generic
configure: error: 
	*** Unable to build an empty module.
			
ERROR (dkms apport): kernel package linux-headers-5.11.0-7612-generic is not supported
Error! Bad return status for module build on kernel: 5.11.0-7612-generic (x86_64)
Consult /var/lib/dkms/zfs/0.8.3/build/make.log for more information.
dpkg: error processing package zfs-dkms (--configure):
 installed zfs-dkms package post-installation script subprocess returned error exit status 10
```

What can I do to be able to use zfs?

**Steps to reproduce (if you know):**

**Expected behavior:**

**Other Notes:**
